adafruit_wiznet5k/adafruit_wiznet5k.py

```diff
@@ -727,21 +727,20 @@ def get_socket(self) -> int:
 
         Cycle through the sockets to find the first available one, if any.
 
-        :return int: The first available socket. Returns 0xFF if no sockets are free.
+        :return int: The first available socket.
+
+        :raises RuntimeError: If no sockets are available.
         """
         if self._debug:
             print("*** Get socket")
 
-        sock = _SOCKET_INVALID
-        for _sock in range(self.max_sockets):
-            status = self.socket_status(_sock)[0]
+        for sock in range(self.max_sockets):
+            status = self.socket_status(sock)[0]
             if status == SNSR_SOCK_CLOSED:
```

Comment:
If the socket isn't used right away, it compares as closed and it will keep allocating socket #0 ad infinitum.

Reply:
Sorry, I know it's draft, but I figured I'd try it out.

Reply:
> If the socket isn't used right away, it compares as closed and it will keep allocating socket #0 ad infinitum.

Thanks. I suspect this has always been the case as `WIZNET5K` does not track which sockets are in use.

I'll have a look at how this might be tracked. Meanwhile I'll close this PR.